app/javascript/crayons/MentionAutocomplete/__tests__/MentionAutocompleteCombobox.test.jsx

```diff
@@ -0,0 +1,110 @@
+import { h } from 'preact';
+import { render, waitFor } from '@testing-library/preact';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import { axe } from 'jest-axe';
+import { MentionAutocompleteCombobox } from '../MentionAutocompleteCombobox';
+
+describe('<MentionAutocomplete />', () => {
+  const mockCoords = {
+    x: 0,
+    y: 0,
+  };
+
+  const mockOnSelect = jest.fn();
+  const mockOnSearchtermChange = jest.fn();
+
+  it('should have no a11y violations when rendered', async () => {
+    const { container } = render(
+      <MentionAutocompleteCombobox
+        onSelect={mockOnSelect}
+        onSearchTermChange={mockOnSearchtermChange}
+        fetchSuggestions={() => Promise.resolve([])}
+        placementCoords={mockCoords}
+      />,
+    );
+
+    const results = await axe(container);
+
+    expect(results).toHaveNoViolations();
+  });
+
+  it('should render', () => {
+    const { container } = render(
+      <MentionAutocompleteCombobox
+        onSelect={mockOnSelect}
+        onSearchTermChange={mockOnSearchtermChange}
+        fetchSuggestions={() => Promise.resolve([])}
+        placementCoords={mockCoords}
+      />,
+    );
+
+    expect(container.innerHTML).toMatchSnapshot();
+  });
+
+  it('should not fetch suggestions with less than three characters', async () => {
```

Comment:
'3 characters' because the first one would be the '@' sign (testing library's user event doesn't support typing that character) 😄